Assets/Scripts/Interactables/GrahamCrackerPart.cs

```diff
@@ -69,6 +69,32 @@ public void MoveAway(float speed)
             StartCoroutine(QueryForInitialPosition(_rigidbody2D.position));
         }
 
+        /// <summary>
+        /// Shakes in place to warn that cracker is about to shut.
+        /// </summary>
+        /// <param name="duration">Duration of the shake.</param>
+        /// <param name="shakeDelay">Delay between shakes.</param>
+        /// <param name="distance">Furthest distance from original position when shaking.</param>
+        public void HandleShake(float duration, float shakeDelay, float distance) =>
+            StartCoroutine(Shake(duration, shakeDelay, distance));
+        
+        private IEnumerator Shake(float duration, float shakeDelay, float shakeDistance)
+        {
+            Vector3 startPos = transform.position;
+        
+            for (float timer = 0; timer < duration; timer += Time.deltaTime)
```

Comment:
hm `timer += Time.deltaTime` looks fishy, especially since we're potentially waiting for `shakeDelay`, maybe `shakeDelay > 0 ? shakeDelay : Time.deltaTime`?